src/rules/at-root-no-redundant-selector/index.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+
+const { utils } = require("stylelint");
+const namespace = require("../../utils/namespace");
+const ruleUrl = require("../../utils/ruleUrl");
+
+const ruleName = namespace("at-root-no-redundant-selector");
+
+function isInterpolated(string) {
+  const interpolationRegex = /#{(.*?)}/g;
+  return interpolationRegex.test(string);
+}
+
+function isAtRootRuleNested(node) {
```

Comment:
You don't have to manually go down the tree recursively, you can use `walkAtRules` with any container node, which inclides AtRule nodes, so.. you can do this instead:

```js
function findAtRoot(node) {
  let atRoot;
  node.walkAtRules('at-root', node => {
    atRoot = node;
    return false;
  });
  return atRoot;
}
```

Reply:
Great suggestion, thank you!!